app/src/org/commcare/utils/CrashUtil.java

```diff
@@ -0,0 +1,56 @@
+package org.commcare.utils;
+
+import android.app.Application;
+import android.content.Context;
+import android.webkit.URLUtil;
+
+import com.crashlytics.android.Crashlytics;
+
+import org.commcare.CommCareApplication;
+import org.commcare.android.logging.ReportingUtils;
+import org.commcare.dalvik.BuildConfig;
+
+import io.fabric.sdk.android.Fabric;
+
+/**
+ * Contains constants and methods used in Crashlytics reporting.
+ *
+ * Created by wpride1 on 3/3/15.
+ */
+public class CrashUtil {
+
+    private static final String APP_VERSION = "app_version";
+    private static final String APP_NAME = "app_name";
+    private static final String DOMAIN = "domain";
+    private static final String DEVICE_ID = "device_id";
+
+    private static boolean crashlyticsEnabled = BuildConfig.USE_CRASHLYTICS;
+
+    public static void reportException(Exception e) {
```

Comment:
I know you were just copying the prior behavior of `AcraUtil`, but I'm confused as to why the only place this is being called is when a `CommCareTask` encounters an exception. @wpride Can you weigh in is at why that was the case for ACRA? I would have thought we would also be reporting things to ACRA in lots of other places, namely in `CommCareExceptionHandler`.

Reply:
@amstone326 kind of spit balling here since I wrote this a while ago and don't remember original intent well, but all actual *crashes* were caught and reported through the '@ReportsCrashes` annotation on `CommCareApplication` removed [here]( https://github.com/dimagi/commcare-android/pull/1796/files#diff-6702335a8e0e670e374872510c46a8b5L111)

I believe this call was necessary only in cases where we were catching and displaying errors to the user *without crashing* but we still wanted to crash to get the ACRA. Its possible there are other places where we should be doing this, or that we should remove this (IE maybe if we think users can debug the issue then we don't need it on our servers).

Reply:
@shubham1g5 should confirm that something like the `ReportsCrashes` annotation exists here that automatically reports all hard crashes

Reply:
So all the hard crashes get automatically intercepted. I have confirmed it by triggering a crash manually. Though the exceptions we are catching won't make it to the Crashlytics without calling this `reportException` explicitly. I am in favour of reporting all exceptions to Crashlytics but since we already send those to HQ not sure if we would want to do that.

Reply:
Cool, sounds fine to me.

Reply:
@shubham1g5 If I'm understanding what you said correctly, that means that any crashes which are caught and stopped [here](https://github.com/dimagi/commcare-android/blob/97bba52eabe2d6b47c606f28beea67f5ecc289b9/app/src/org/commcare/utils/CommCareExceptionHandler.java#L36-L39) won't end up on Crashlytics as of now. If that's correct, can we just add an explicit call to `reportException()` within that if statement as well?